src/core/ComposableElement.php

```diff
@@ -134,6 +133,7 @@ protected function init(): void {
       }
     }
     }
+    /*HH_FIXME[4110] $new_children was appended on line 132 with a nonnull element*/
```

Comment:
1. Line numbers are likely to get out of sync so I would recommend against mentioning them (just "appended above").

2. I suspect the "potentially nullable" actually comes from the "if is traversable, append everything" since you haven't asserted that someone didn't pass in an array with nulls.

Reply:
The hover type of $xhp in 
```HACK
else if (!($xhp is Traversable<_>)) {
  $new_children->add($xhp);
}
```
remains XHPChild under my HHVM4.14-ish typechecker. I'll update the message and not add the extra typechecks to make the typechecker happy. `// This function has been micro-optimized` is making me back off. If this is something that we should typecheck, just let me know.